internal/campaigns/repo_fetcher_test.go

```diff
@@ -0,0 +1,203 @@
+package campaigns
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/sourcegraph/src-cli/internal/api"
+	"github.com/sourcegraph/src-cli/internal/campaigns/graphql"
+)
+
+func TestRepoFetcher_Fetch(t *testing.T) {
+	workspaceTmpDir := func(t *testing.T) string {
+		testTempDir, err := ioutil.TempDir("", "executor-integration-test-*")
+		if err != nil {
+			t.Fatal(err)
+		}
+		t.Cleanup(func() { os.Remove(testTempDir) })
+
+		return testTempDir
+	}
+
+	repo := &graphql.Repository{
+		ID:            "src-cli",
+		Name:          "github.com/sourcegraph/src-cli",
+		DefaultBranch: &graphql.Branch{Name: "main", Target: graphql.Target{OID: "d34db33f"}},
+	}
+
+	archive := mockRepoArchive{
+		repo: repo,
+		files: map[string]string{
+			"README.md": "# Welcome to the README\n",
+		},
+	}
+
+	t.Run("success", func(t *testing.T) {
+		requestsReceived := 0
+		callback := func(_ http.ResponseWriter, _ *http.Request) {
+			requestsReceived++
+		}
+
+		ts := httptest.NewServer(newZipArchivesMux(t, callback, archive))
+		defer ts.Close()
+
+		var clientBuffer bytes.Buffer
+		client := api.NewClient(api.ClientOpts{Endpoint: ts.URL, Out: &clientBuffer})
+
+		rf := &repoFetcher{
+			client:     client,
+			dir:        workspaceTmpDir(t),
+			deleteZips: false,
+		}
+
+		rz, err := rf.Fetch(context.Background(), repo)
+		if err != nil {
+			t.Errorf("unexpected error: %s", err)
+		}
+
+		wantZipFile := "github.com-sourcegraph-src-cli-d34db33f.zip"
+		ok, err := dirContains(rf.dir, wantZipFile)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !ok {
+			t.Fatalf("temp dir doesnt contain zip file")
+		}
+
+		// We can't use path.Join() here because it hardcodes / as the path
```

Comment:
What about filepath.Join?

Reply:
I... uh, don't know why I forgot that existed. Thanks!